modular_skyrat/master_files/code/modules/antagonists/traitor/objectives/kill_pet.dm

```diff
@@ -0,0 +1,100 @@
+	var/obj/item/card/emag/oneshot/oneshot_emag
+
+
+////
+//	Objective overwrites
+/datum/traitor_objective/kill_pet/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	. = ..()
+	// Dust Poppy the safety inspector
+	if(istype(target_pet, /mob/living/simple_animal/pet/poppy))
+		name = "Dust the engineering department's esteemed safety inspector and beloved pet, Poppy"
+		description = "A couple of troublemakers in the engineering department have spilled the milk, make them and their colleagues pay for the consequences by throwing Poppy the safety inspector into the supermatter engine "
+		// Cleaning up the original success_signals which are `list(COMSIG_PARENT_QDELETING, COMSIG_LIVING_DEATH)`
+		for(var/datum/component/traitor_objective_register/old_objective as anything in GetComponents(/datum/component/traitor_objective_register))
+			if(old_objective.target == target_pet)
+				qdel(old_objective)
+		// Adding our own signal component, targeting `target_pet`
+		AddComponent(/datum/component/traitor_objective_register, target_pet, \
+			succeed_signals = list(COMSIG_PARENT_QDELETING)) // Until dusting gets its own component, this has to make do
+
+	// Emag E-N so it overloads
+	if(istype(target_pet, /mob/living/simple_animal/pet/dog/corgi/borgi))
+		name = "Emag the roboticist's most-prized borgi, E-N"
+		description = "We received an untracable contract from someone addressed 'ianfan2489@solgov.nt', they want the robotic clone of Ian dead- and it has to be dramatic. We will supply you with a cryptographic sequencer from our basic loadout options to Shortcircuit E-N's safety, it will give you a single attempt to use it. Oh, and remember to run like hell when you do... "
+
+		for(var/datum/component/traitor_objective_register/old_objective as anything in GetComponents(/datum/component/traitor_objective_register))
+			if(old_objective.target == target_pet)
+				qdel(old_objective)
+
+		AddComponent(/datum/component/traitor_objective_register, target_pet, \
+			succeed_signals = list(COMSIG_ATOM_EMAG_ACT))
+
+////
+//	Objective item (for emagging E-N)
+/obj/item/card/emag/oneshot
+	name = "cryptographic sequencer"
+	special_desc_requirement = EXAMINE_CHECK_JOB
+	special_desc_jobs = list(JOB_DETECTIVE, JOB_HEAD_OF_SECURITY)
+	special_desc = "Upon inspection you can instantly tell this is a real cryptographic sequencer commonly traded in bulk for cheap at countless blackmarkets. They are known for their unreliability and breaking after just one use from their shoddy construction."
+	///How many uses does it have left?
+	var/charges = 1
+	///Who summoned this?
+	var/caller
+
+/obj/item/card/emag/oneshot/examine(mob/user)
+	. = ..()
+	if(user == caller)
+		. += span_notice("It looks cheapo, they did say it gives just one shot...")
+	else
+		. += span_notice("It looks flimsy and identical to the \"DonkCo\" toy.")
+
+/obj/item/card/emag/oneshot/can_emag(atom/target, mob/user)
+	if(charges <= 0)
```

Comment:
You probably also want to make sure it can only be used on E-N if you do this, otherwise they could use it to get a free one-time emag and just soft-lock themselves out of the objective.

Reply:
At which point you say "don't use it on other stuff then"

Reply:
If a traitor loses the objective given emag they face two options: to give the objective up and get penalized in TC, or to buy a real emag, costing AFAIK an equal amount of TC.

Reply:
Its held in your uplink until you're ready to spawn it, no real excuse except PEBCAK if you fuck it up.